core/ecschema-metadata/src/UnitConversion/UnitConversion.ts

```diff
@@ -0,0 +1,87 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { Constant, Unit } from "../ecschema-metadata";
+import { SchemaItemType } from "../ECObjects";
+import { Float } from "./Float";
+
+/**
+ * Class used for storing calculated conversion between two Units [[UnitConverter.calculateConversion]] and converting values from one Unit to another [[UnitConverter.evaluate]]
+ * @alpha
+ */
+export class UnitConversion {
+  /** @internal */
+  constructor(public readonly factor: number = 1.0, public readonly offset: number = 0.0) {}
+
+  /**
+   * Converts x using UnitConversion
+   * @param x Input magnitude to be converted
+   * @returns Output magnitude after conversion
+   */
+  public evaluate(x: number): number {
+    return this.factor * x + this.offset;
+  }
+
+  /**
+   * Used to invert source's UnitConversion so that it can be composed with target's UnitConversion cleanly
+   * @internal
+   */
+  public inverse(): UnitConversion {
+    const inverseFactor = 1.0 / this.factor;
+    return new UnitConversion(inverseFactor, -this.offset * inverseFactor);
+  }
+
+  /**
+   * Combines two UnitConversions
+   * Used to combine source's UnitConversion and target's UnitConversion for a final UnitConversion that can be evaluated
+   * @internal
+   */
+  public compose(conversion: UnitConversion): UnitConversion {
+    return new UnitConversion(
+      this.factor * conversion.factor,
+      conversion.factor * this.offset + conversion.offset
+    );
+  }
+
+  /**
+   * Multiples two UnitConversions together to calculate factor during reducing
+   * @internal
+   */
+  public multiply(conversion: UnitConversion): UnitConversion {
+    if (Float.equals(conversion.offset, 0.0) && Float.equals(this.offset, 0.0))
+      return new UnitConversion(this.factor * conversion.factor, 0.0);
+
+    throw new Error("Cannot multiply two maps with non-zero offsets");
+  }
+
+  /**
+   * Raise UnitConversion's factor with power exponent to calculate factor during reducing
+   * @internal
+   */
+  public raise(power: number): UnitConversion {
+    if (Float.equals(1.0, power))
+      return new UnitConversion(this.factor, this.offset);
+    else if (Float.equals(0.0, power))
+      return new UnitConversion(1.0, 0.0);
+
+    if (Float.equals(this.offset, 0.0))
+      return new UnitConversion(this.factor ** power, 0.0);
+
+    throw new Error("Cannot raise map with non-zero offset");
+  }
+
+  /** @internal */
+  public static identity = new UnitConversion();
```

Comment:
Don't quite understand why this is needed in the places it's used...

Reply:
That creates a UnitConversion with a factor of 1 and offset of 0, which is like a non-meaningful base value. The factor will be used multiplicatively, so anything that multiplies by 1 will give back its identity. The offset is used additively, and anything that adds 0 will give back itself.

Reply:
Unless you mean I should just use `new UnitConversion()` inline in the places identity is used.